package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -60,10 +84,112 @@ def to_json(self) -> str:
+        databases = fs.glob(f"{self._remote_path}/*.db")
+        databases_location = []
+        # Download each database to a local filepath
+        for database in databases:
+            database_name = Path(database).name
+            with fs.open(database, "rb") as file:
+                db_data = file.read()
+                db_loc = f"{self._path}/{database_name}"
+                with open(db_loc, "wb") as temp_db:
+                    temp_db.write(db_data)
+            databases_location.append(db_loc)
+        # Return the list of local filepaths
+        return databases_location
+
+    def _merge(self, databases_location: List[str]):
+        """Merges all the session store databases stored at the specified locations into the user's local session_store.db
+
+        Notes:
+        - This method uses multiple SQLAlchemy engines to connect to the user's session_store.db and to all the other downloaded dbs.
+        - It is assumed that all the databases share the same schema.
+        - In the version 1.0 - we only merge the runs table which contains all the experiments.
+        - The downloaded database files are deleted after it's runs are merged with the user's local session_store.db
+
+        Args:
+            database_location:  A list of local filepath in string format for all the databases
+
+        """
+
+        # Connect to the user's local session_store.db
+        engine, session_class = create_db_engine(self.location)
+        Base.metadata.create_all(bind=engine)
+        database = next(get_db(session_class))
+
+        temp_engine = None
+        # Iterate through each downloaded database
+        for db_loc in databases_location:
+            if temp_engine:
+                temp_engine.dispose()
+            # Open a connection to the downloaded database
+            temp_engine = create_engine(f"sqlite:///{db_loc}")
+            with temp_engine.connect() as database_conn:
+                db_metadata = MetaData()
+                db_metadata.reflect(bind=temp_engine)
+                # Merge data from the 'runs' table
+                for table_name, table_obj in db_metadata.tables.items():
+                    if table_name == "runs":
+                        data = database_conn.execute(table_obj.select()).fetchall()
+                        for row in data:
+                            row_dict = row._asdict()
+                            session_store_data = RunModel(**row_dict)
+                            try:
+                                database.add(session_store_data)
+                                database.commit()
+                            except Exception as e:
+                                database.rollback()
+                                logging.exception(f"Failed to add runs to the {database}: {e}")
+            # Close the connection to the downloaded database and delete it
+            temp_engine.dispose()
+            os.remove(db_loc)
```

Comment:
Do we actually need to remove the databases? Might just be easier to leave them there, since it's one less thing that could go wrong and should make things easier to debug, and it won't use up much disk space.

Reply:
This was just to keep the user's project directory clean. I mean these were only temp databases.